doctr/transforms/functional/tensorflow.py

```diff
@@ -52,6 +52,8 @@ def rotate(
     if expand:
         exp_shape = compute_expanded_shape(img.shape[:-1], angle)
         h_pad, w_pad = int(math.ceil(exp_shape[0] - img.shape[0])), int(math.ceil(exp_shape[1] - img.shape[1]))
+        if min(h_pad, w_pad) < 0:
+            h_pad, w_pad = int(math.ceil(exp_shape[1] - img.shape[0])), int(math.ceil(exp_shape[0] - img.shape[1]))
```

Comment:
mmmh ? but if one of those is negative, it means that `compute_expanded_shape` has to be fixed. Over here, that shouldn't happen no ?

Reply:
If you rotate more than 45 degrees a rectangular document, you may end up with a expanded width smaller than the previous width (because basically you switch the axes), in that case one of the 2 pad should be negative and I think we just have to switch axes for exp_shape what do you think ? In pytorch it is hidden in the PIL transform

Reply:
Oh I see, yes the padding should be applied to the opposite axis then I agree!